nixos/modules/services/web-apps/nextcloud-backup.nix

```
@@ -0,0 +1,196 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.nextcloud;
+  backupRsyncCommand = "${lib.getExe pkgs.rsync} -rlt --del --perms";
+  restoreRsyncCommand = "${backupRsyncCommand} --chmod=u+rwX --chown=nextcloud:nextcloud";
+  dbBackupRestore =
+    if cfg.config.dbtype == "sqlite" then
+      {
+        # The sqlite database is automatically backed up since it's contained in the datadir
+        backupFileName = "data/${cfg.config.dbname}.db";
+        backupCommand = "";
+        restoreCommand = "";
+      }
+    else if cfg.config.dbtype == "pgsql" then
+      let
+        backupFileName = "database-pgsql.bak";
+        pg_dump = "${config.services.postgresql.package}/bin/pg_dump";
+        pg_restore = "${config.services.postgresql.package}/bin/pg_restore";
+        setPassword =
+          if cfg.config.dbpassFile != null then
```

You can simplify this:
```suggestion
          lib.optionalString (cfg.config.dbpassFile != null) "export PGPASSWORD=..."
```